firefox-ios/firefox-ios-tests/Tests/ClientTests/Coordinators/Mocks/MockStoreForMiddleware.swift

```diff
@@ -49,7 +51,11 @@ class MockStoreForMiddleware<State: StateType>: DefaultDispatchStore {
         // TODO: if you need it
     }
 
+    /// We implemented the lock to ensure that this is thread safe
+    /// since actions can be dispatch in concurrent tasks
```

Comment:
@lmarceau Carson helped me address some unit test issues I was facing, and mentioned tag you / Orla to confirm if this aligns with our unit tests guidelines. Another proposal Carson mentioned was to possibly use actors (which I'll investigate in a separate ticket). Let me know if you have any thoughts or concerns with this change. 

@Cramsden feel free to add additional context!

cc: @OrlaM to review when she feels better + returns from PTO

Reply:
cc: @ih-codes in case you were interested since this mock store middleware started with you!